pymilvus/orm/connections.py

```diff
@@ -211,8 +218,16 @@ def remove_connection(self, alias: str):
         if not isinstance(alias, str):
             raise ConnectionConfigException(message=ExceptionsMessage.AliasType % type(alias))
 
-        self.disconnect(alias)
+        # TODO: does order matter?
+        # original sync implementation was
+        #     self.disconnect(alias)
+        #     self._alias.pop(alias, None)
```

Comment:
there is a possibility that it could be because a disconnect was unsuccessful which raised an error such that the alias wasn't popped in that errored event. But not sure.

Reply:
I assume its as you have stated, make sure disconnected before removing so that we are sure there are no ghost connections left over.

Reply:
Could you describe an example scenario?

Current implementation of `.disconnect()` actually removes a GrpcHandler first and *then* calls `.close()`:
https://github.com/milvus-io/pymilvus/blob/94c210c7f0385d0d4c76190b74cde40a491fc966/pymilvus/orm/connections.py#L203

In case that some exception is raised from `GrpcHandler.close()`, the grpc handler instance will be "lost" anyway (removed from `._connected_alias` dict). Is it meaningful to keep a corresponding value in `self._alias` in that case (instead of removing as requested by user)?

Anyway, I've updated the code to keep original statements order. But it may turn out that we actually must use some asyncio synchronization primitives for reliable behavior in case of errors.